woocommerce/woocommerce-admin
# Reduce order status change delay in Orders Panel.

Fixes #2592 and #2309.

This PR seeks to reduce/eliminate the delay in order status transitions manifesting in the Orders Panel.

It does so by using the core WooCommerce Orders endpoint to determine if there are actionable orders, as well as trusting the core endpoint data over the reporting table data - both endpoints are hit and the data merged, with the core endpoint overriding the reporting endpoint. This hopefully ensures that we can catch order status (and date, etc) changes in any "direction" - processing to completed, or completed to pending, etc.

Additionally, this PR removes the default date range for the `/reports/orders` endpoint (since it's not present in the core endpoint, and we need parity). This solves #2309.

### Detailed test instructions:

- Have an order that has an "actionable" status and shows up in the activity panel.
- Edit the order to a non-actionable status, i.e. mark it as complete.
- Note after the POST of the edit order form is completed, the order is no longer in the panel.
- Navigate to the wc-admin dashboard, verify the order is no longer in the panel.



### Changelog Note:

Fix: completed orders lingering in activity panel.
